qemu/tests/driver_load.py

```diff
@@ -21,73 +22,99 @@ def run(test, params, env):
+            raise exceptions.TestFail("failed to load driver, %s" % output)
+
+    def unload_driver(cmd, driver_id):
+        """
+        Unload driver
+        :param cmd: Driver unload cmd
+        :param driver_id: Driver id in windows guest
+        """
+        nic_index = len(vm.virtnet) - 1
+        session = vm.wait_for_login(nic_index=nic_index)
         if params["os_type"] == "windows":
             cmd = cmd.replace("DRIVER_ID", driver_id)
 
         status, output = session.cmd_status_output(cmd)
+        session.close()
         if status != 0:
-            raise error.TestFail("failed to unload driver, %s" % output)
-        if params["os_type"] == "windows":
-            if "device(s) disabled" not in output:
-                raise error.TestFail("failed to unload driver, %s" % output)
-
-    def check_driver(session, cmd, pattern):
+            if "reboot" in output:
+                vm.reboot()
+                session.close()
+            else:
+                raise exceptions.TestFail("failed to unload driver, %s" %
+                                          output)
+
+    def get_driver_id(cmd, pattern):
+        """
+        Get driver id from guest
+        :param cmd: cmd to get driver info
+        :param pattern: pattern to filter driver id
+        """
+        nic_index = len(vm.virtnet) - 1
+        session = vm.wait_for_login(nic_index=nic_index)
         output = session.cmd_output(cmd)
         driver_id = re.findall(pattern, output)
         if not driver_id:
-            raise error.TestFail("Didn't find driver info from guest %s"
-                                 % output)
+            raise exceptions.TestFail("Didn't find driver info from guest %s"
+                                      % output)
 
         driver_id = driver_id[0]
         if params["os_type"] == "windows":
             driver_id = '^&'.join(driver_id.split('&'))
+        session.close()
         return driver_id
 
-    error.context("Try to log into guest.", logging.info)
+    error_context.context("Try to log into guest.", logging.info)
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
     timeout = float(params.get("login_timeout", 240))
-    # Use the last nic for send driver load/unload command
-    nic_index = len(vm.virtnet) - 1
-    session = vm.wait_for_login(nic_index=nic_index, timeout=timeout)
 
     driver_id_cmd = params["driver_id_cmd"]
     driver_id_pattern = params["driver_id_pattern"]
     driver_load_cmd = params["driver_load_cmd"]
     driver_unload_cmd = params["driver_unload_cmd"]
 
-    devcon = params.get("devcon")
-    if devcon:
-        error.context("Copy devcon.exe from winutils.iso to C:\\")
+    if params.get("devcon"):
+        # Use the last nic for send driver load/unload command
+        nic_index = len(vm.virtnet) - 1
+        session = vm.wait_for_login(nic_index=nic_index, timeout=timeout)
+        error_context.context("Copy devcon.exe from winutils.iso to C:\\")
```

Comment:
Shall we add the parameter "logging.info" to the function? As I cannot find the step description from the log.